### PR TITLE
Remove table backup/restore functionality.

### PR DESCRIPTION
**Pull request by jpinner-lyft**
The backup and restore methods are not scalable and store data in a custom format:
`(<serialized-hash-key>, {"range_key": <serialized-range-key>, "attributes": <dynamodb-attribute-values>})`

### CODE REVIEW

File: tests/test_discriminator.py [hunk 1]
```diff
@@ -45,14 +45,12 @@ def test_serialize(self):
         dtm.value = StringValue(name='foo', value='Hello')
         dtm.values = [NumberValue(name='bar', value=5), RenamedValue(name='baz', value='World')]
         assert dtm._serialize() == {
```
Comment (ikonst) [inline tests/test_discriminator.py:47]: Should we call out that the format is changing? I know at least one piece of code which it'll disrupt (luckily I could increase the cache version there along with the PynamoDB update).

Comment (jpinner-lyft) [reply tests/test_discriminator.py:47]: I'll put a release note in -- I was also planning in the JSON pr to make the serialize/deserialize methods public.

File: pynamodb/models.py [hunk 2]
```diff
@@ -13,7 +12,7 @@
 from pynamodb.expressions.update import Action
 from pynamodb.exceptions import DoesNotExist, TableDoesNotExist, TableError, InvalidStateError, PutError
 from pynamodb.attributes import (
-    Attribute, AttributeContainer, AttributeContainerMeta, MapAttribute, TTLAttribute, VersionAttribute
```
Comment (jmphilli) [inline pynamodb/models.py:16]: interesting to see this was unused - i wonder if we should include a pre-commit linter or something going forward

**Comment by jmphilli** [comment]
+1

**Review by jmphilli — APPROVED** [review]

**Pull request merged**